chain/src/chain.rs

```diff
@@ -2097,7 +2097,7 @@ impl ChainReader for BlockChain {
             let state_key = match access_path.path {
                 DataPath::Code(module_name) => StateKey::module(&access_path.address, &module_name),
                 DataPath::Resource(struct_tag) => {
-                    StateKey::resource(&access_path.address, &struct_tag)?
```

Comment:
Why should anyhow::Result be removed from Statekey::resource ?

Reply:
In the beginning, I want to keep it same with other Statekey type, like resource_group/module. It should be OK to unwrap it immediately to remove the Result wrapper and the only error should occur when serializing StateKey.

Reply:
Will do double check with Upstream codebase.